src/coreclr/ilasm/writer.cpp

```diff
@@ -341,11 +340,11 @@ HRESULT Assembler::CreateExportDirectory()
     // Allocate buffer for tables
     for(i = 0, L=0; i < Nentries; i++) L += 1+(unsigned)strlen(m_EATList.PEEK(i)->szAlias);
     exportDirDataSize = Nentries*5*sizeof(WORD) + L + Ldllname;
-    exportDirData = new BYTE[exportDirDataSize];
+    NewArrayHolder<BYTE> exportDirData(new BYTE[exportDirDataSize]);
     memset(exportDirData,0,exportDirDataSize);
 
     // Export address table
-    DWORD*  pEAT = (DWORD*)exportDirData;
+    DWORD*  pEAT = (DWORD*)(BYTE*)exportDirData;
```

Comment:
This isn't a change in this PR, but a `DWORD` (`unsigned int` on my machine) might have a different alignment than a `BYTE` (`char` on my machine). Could this fail?

Reply:
malloc on Linux returns 8 byte aligned memory for 32 bit systems and 16 byte aligned ones on 64 bit systems. See https://www.gnu.org/software/libc/manual/html_node/Aligned-Memory-Blocks.html. I assume new would behave the same way.